packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputDate.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { unstable_useId as useId } from '@mui/material/utils';
+import { GridLoadIcon } from '../../icons/index';
+import { GridFilterInputValueProps } from './GridFilterInputValueProps';
+
+export interface GridTypeFilterInputDateProps extends GridFilterInputValueProps {
+  type?: 'date' | 'datetime-local';
+}
+
+export const SUBMIT_FILTER_STROKE_TIME = 500;
+
+function GridFilterInputDate(props: GridTypeFilterInputDateProps & TextFieldProps) {
+  const { item, applyValue, type, apiRef, focusElementRef, ...others } = props;
+
+  const filterTimeout = React.useRef<any>();
+  const [filterValueState, setFilterValueState] = React.useState(item.value ?? '');
+  const [applying, setIsApplying] = React.useState(false);
+  const id = useId();
+
+  const onFilterChange = React.useCallback(
+    (event) => {
+      const value = event.target.value;
+
+      clearTimeout(filterTimeout.current);
+      setFilterValueState(String(value));
+
+      setIsApplying(true);
+      filterTimeout.current = setTimeout(() => {
+        applyValue({ ...item, value });
+        setIsApplying(false);
+      }, SUBMIT_FILTER_STROKE_TIME);
+    },
+    [applyValue, item],
+  );
+
+  React.useEffect(() => {
```

Comment:
We miss cleaning the timeout during unmount.

https://github.com/mui-org/material-ui-x/blob/8ffe04e6efe09b17daece526d116df5eff5aa0bd/packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputValue.tsx#L82-L86